### ISSUE TITLE
Ignore .svn directories when creating schemas from annotated classes

### ISSUE THREAD

**Issue by jkraushaar** [mojohaus/jaxb2-maven-plugin#16]
I am generating schema files from annotated classes using schemagen goal. When executed on CI server, I get the following error:

[ERROR] Failed to execute goal org.codehaus.mojo:jaxb2-maven-plugin:2.1:schemagen (schemagen) on project example-project: Execution schemagen of goal org.codehaus.mojo:jaxb2-maven-plugin:2.1:schemagen failed: syntax error @[1,1] in file:/tmp/jenkins-slaves/tests/workspace/example-project/src/main/java/.svn/all-wcprops -> [Help 1]

It seems that jaxb2-maven-plugin also includes files in .svn directory.

System uses svn 1.6.11.

**Comment by lennartj** [comment]
Use a Filter to exclude everything below the .svn directory:

````xml
                <schemaSourceExcludeFilters>
                    <myExcludes implementation="org.codehaus.mojo.jaxb2.shared.filters.pattern.PatternFileFilter">
                        <patterns>
                            <pattern>jaxb\.index</pattern>
                            <pattern>\.txt</pattern>
                            <pattern>\.svn.*</pattern>
                        </patterns>
                    </myExcludes>
                </schemaSourceExcludeFilters>
````

This is documented as an example on the http://www.mojohaus.org/jaxb2-maven-plugin/example_schemagen_basic.html page.

Also, the upcoming 2.2 release prunes a few bugs regarding source selection. I could recommending upgrading - as soon as the 2.2 release is out.

**Comment by lautou** [comment]
This issue is still present in version 2.2. We still must use the filter as described above.

**Comment by fusiongyro** [comment]
This is still an issue in 2.3.1. You should probably skip hidden files (files starting with .) by default.

**Comment by lennartj** [comment]
I would also expect directories starting with '.' to be considered hidden.
I'll fix it in 2.4
